Automattic/wp-calypso
# Atomic Hosting: Restore wp-config.php

**DO NOT MERGE**

Requires D35302-code

#### Changes proposed in this Pull Request

Displays a notice in the Atomic Hosting panel when the `wp-config-missing` query param is present on the URL. The notice includes a link that restores the `wp-config.php` file.

<img width="1108" alt="Screen Shot 2019-11-13 at 12 45 42" src="https://user-images.githubusercontent.com/1233880/68761117-ade78d00-0613-11ea-8b82-159129e0b60e.png">

We currently display an error page when a user visits an Atomic site with the `wp-config.php` file missing:

![image](https://user-images.githubusercontent.com/1233880/68761222-ed15de00-0613-11ea-96c3-fa17acff8cfe.png)

Once this PR is out, there will be some work in Atomic that will change the wording and the link so it redirects to `/hosting-admin/:site?wp-config-missing` giving the user a chance to self-restore the file.

#### Testing instructions

* Apply D35302-code and sandbox the API.
* Connect to the SFTP of your Atomic site and delete the `wp-config.php` file.
* Visit the frontend of your AT site and note the "Missing wp-config.php" error.
* In Calypso, go to `/hosting-admin/:site/?wp-config-missing`.
* Make sure the page displays an error notice with a restore file link action.
* Click on "Restore file".
* Verify the `wp-config-missing` has been removed when the success notification is displayed.
* Verify the `wp-config.php` file has been restored by visiting the frontend of your AT site.

Fixes #37372
